build.sh

```diff
@@ -1,8 +1,89 @@
 #!/usr/bin/env bash
 
+set -euo pipefail
+IFS=$'\n\t'
+
+usage() {
+    echo "usage: $0 [options]"
+    echo "options:"
+    echo "  --with-packages <dir>              use the specified archive of source-built packages to build."
+    echo "  --with-sdk <dir>                   use the specified SDK to build."
+}
+
+function abspath() {
+  # $1 : relative filename
+  parentdir=$(dirname "$1")
+
+  if [ -d "$1" ]; then
+      echo "$(cd "$1" && pwd)"
+  elif [ -d "${parentdir}" ]; then
+    echo "$(cd "${parentdir}" && pwd)/$(basename "$1")"
+  fi
+}
+
 __scriptpath=$(cd "$(dirname "$0")"; pwd -P)
 __log_path="$__scriptpath/artifacts/log"
 
+CUSTOM_SDK_DIR=''
+CUSTOM_SOURCE_BUILT_PACKAGES=''
+
+while :; do
+    if [ $# -le 0 ]; then
+        break
+    fi
+
+    lowerI="$(echo $1 | awk '{print tolower($0)}')"
+    case $lowerI in
+        --with-sdk)
+            CUSTOM_SDK_DIR="$2"
+            if [ ! -d "$CUSTOM_SDK_DIR" ]; then
+                echo "Custom SDK directory '$CUSTOM_SDK_DIR' does not exist"
+            fi
+            if [ ! -x "$CUSTOM_SDK_DIR/dotnet" ]; then
+                echo "Custom SDK '$CUSTOM_SDK_DIR/dotnet' not found or not executable"
+            fi
+            shift
+            ;;
+        --with-packages)
+            CUSTOM_SOURCE_BUILT_PACKAGES="$2"
+            if [ ! -f "$CUSTOM_SOURCE_BUILT_PACKAGES" ]; then
+                echo "Custom source-built packages file '$CUSTOM_SOURCE_BUILT_PACKAGES' does not exist"
+                exit 1
+            fi
+            CUSTOM_SOURCE_BUILT_PACKAGES="$(abspath "$CUSTOM_SOURCE_BUILT_PACKAGES")"
+            shift
+            ;;
+        -?|-h|--help)
+            usage
+            exit 0
+            ;;
+        *)
+            echo "Unrecognized argument '$1'"
+            usage
+            exit 1
+            ;;
+    esac
+
+    shift
+done
+
+if [ -n "${CUSTOM_SDK_DIR}" ]; then
+    rm -rf $__scriptpath/.dotnet
```

Comment:
Missing quotes for spaces (and below).

Reply:
Should be fixed in https://github.com/dotnet/source-build-reference-packages/pull/102